ibc-apps/ics721-nft-transfer/types/src/error.rs

```diff
@@ -16,80 +15,45 @@ pub enum NftTransferError {
     ContextError(ContextError),
     /// invalid identifier: `{0}`
     InvalidIdentifier(IdentifierError),
-    /// invalid URI: `{uri}`, validation error: `{validation_error}``
+    /// invalid URI: `{uri}`, validation error: `{validation_error}`
     InvalidUri {
         uri: String,
         validation_error: http::uri::InvalidUri,
     },
-    /// destination channel not found in the counterparty of port_id `{port_id}` and channel_id `{channel_id}`
-    DestinationChannelNotFound {
+    /// invalid json data: `{description}`
+    InvalidJsonData { description: String },
+    /// invalid trace `{trace_path}`: `{description}`
+    InvalidTrace {
+        trace_path: String,
+        description: String,
+    },
+    /// missing destination channel `{channel_id}` on port `{port_id}`
+    MissingDestinationChannel {
         port_id: PortId,
         channel_id: ChannelId,
     },
-    /// base class ID is empty
+    /// empty base class ID
     EmptyBaseClassId,
-    /// invalid prot id n trace at position: `{pos}`, validation error: `{validation_error}`
-    InvalidTracePortId {
-        pos: u64,
-        validation_error: IdentifierError,
-    },
-    /// invalid channel id in trace at position: `{pos}`, validation error: `{validation_error}`
-    InvalidTraceChannelId {
-        pos: u64,
-        validation_error: IdentifierError,
-    },
-    /// trace length must be even but got: `{len}`
-    InvalidTraceLength { len: u64 },
-    /// no token ID
-    NoTokenId,
-    /// invalid token ID
-    InvalidTokenId,
-    /// duplicated token IDs
-    DuplicatedTokenIds,
-    /// The length of token IDs mismatched that of token URIs or token data
-    TokenMismatched,
-    /// invalid json data
-    InvalidJsonData,
-    /// the data is not in the JSON format specified by ICS-721
-    InvalidIcs721Data,
-    /// expected `{expect_order}` channel, got `{got_order}`
-    ChannelNotUnordered {
-        expect_order: Order,
-        got_order: Order,
-    },
+    /// empty token ID
+    EmptyTokenId,
+    /// mismatched number of token IDs: expected `{expected}`, actual `{actual}`
+    MismatchedNumberOfTokenIds { expected: usize, actual: usize },
+    /// mismatched channel orders: expected `{expected}`, actual `{actual}`
+    MismatchedChannelOrders { expected: Order, actual: Order },
+    /// mismatched port IDs: expected `{expected}`, actual `{actual}`
+    MismatchedPortIds { actual: PortId, expected: PortId },
```

Comment:
To be consistent, should the `expected` field comes before `actual`?

Reply:
You mean just make sure the order in which we specify `expected` and `actual` in the variants match?

Reply:
[250766f](https://github.com/cosmos/ibc-rs/pull/1318/commits/250766f61141945b47b24c7b04cd6d7c82fb07b6)